flash_examples/flash_components/custom_flash_datasets.py

```diff
@@ -0,0 +1,249 @@
+#           keys="input",
+#           transform=Compose(
+#                ToTensor()
+#                RandomRotation(degrees=[-45.0, 45.0], interpolation=nearest, expand=False, fill=0)),
+#        TransformPlacement.COLLATE: default_collate,
+#    },
+# )
+
+
+validation_dataset = MultipleFoldersImageDataset.from_data(
+    VAL_FOLDERS, RunningStage.VALIDATING, transform=CustomDataTransform.RANDOM_ROTATION
+)
+
+print(validation_dataset.transform)
+# Out:
+# FlashRandomRotationTransform(
+#    running_stage=validate,
+#    transform={
+#        TransformPlacement.PER_SAMPLE_TRANSFORM: ApplyToKeys(keys="input", transform=ToTensor()),
+#        TransformPlacement.COLLATE: default_collate,
+#    },
+# )
+
+print(train_dataset[0])
+# Out:
+# {
+#   <DefaultDataKeys.INPUT: 'input'>: tensor([...]),
+#   <DefaultDataKeys.TARGET: 'target'>: 0,
+#   <DefaultDataKeys.METADATA: 'metadata'>: (500, 375)
+# }
+
+#############################################################################################
+#                        Step 4 / 5: Create an FlashDatasetsContainer                       #
+#                                                                                           #
+# The `FlashDatasetsContainer` class is a collection of DataSource which can be built out   #
+# with `class_method` constructors.                                                         #
+# The `FlashDatasetsContainer` requires a FlashRegistry `data_sources_registry`             #
+# class attributes. By creating a `from_multiple_folders`, we can easily create a           #
+# constructor taking the folders paths and by using the `cls.from_data_source`              #
+# with `CustomDataFormat.MULTIPLE_FOLDERS`, it would indicate the parent class the          #
+# associated `DataSource` is our newly implemented one. The extra arguments are the         #
+# `{stage}_data` to be passed to the `{stage}_load_data` from the associated                #
+# `MultipleFoldersImage`.                                                                   #
+#                                                                                           #
+#############################################################################################
+
+
+class ImageClassificationContainer(FlashDatasetsContainer):
+
+    data_sources_registry = FlashRegistry("image_classification_loader")
+    default_data_source = CustomDataFormat.MULTIPLE_FOLDERS
+
+    @classmethod
+    def from_multiple_folders(
+        cls,
+        train_folders: Optional[List[str]] = None,
+        val_folders: Optional[List[str]] = None,
+        test_folders: Optional[List[str]] = None,
+        predict_folder: Optional[str] = None,
+    ):
+        return cls.create_flash_datasets(
+            CustomDataFormat.MULTIPLE_FOLDERS, train_folders, val_folders, test_folders, predict_folder
+        )
+
+
+ImageClassificationContainer.register_data_source(MultipleFoldersImageDataset, CustomDataFormat.MULTIPLE_FOLDERS)
```

Comment:
Calling `register_data_source` and then passing in a dataset feels unintuitive. If I already have a dataset why do I then need a container for it? As a contributor, I know that there has to be some way to have different sources of data for the dataset, but the flow of defining a dataset first and then telling Flash where to get the data for it is a bit backward in my opinion.